log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationFactory.java

```diff
@@ -137,71 +119,15 @@ public ConfigurationFactory() {
      */
     private static final String CLASS_PATH_SCHEME = "classpath";
 
-    private static final String OVERRIDE_PARAM = "override";
-
-    private static volatile List<ConfigurationFactory> factories;
-
-    private static volatile ConfigurationFactory configFactory;
-
-    protected final StrSubstitutor substitutor = new ConfigurationStrSubstitutor(new Interpolator());
-
-    private static final Lock LOCK = new ReentrantLock();
-
-    private static final String HTTPS = "https";
-    private static final String HTTP = "http";
-
     private static final String[] PREFIXES = {"log4j2.", "log4j2.Configuration."};
 
-    private static volatile AuthorizationProvider authorizationProvider;
-
-    /**
-     * Returns the ConfigurationFactory.
-     * @return the ConfigurationFactory.
-     */
+    @Deprecated(since = "3.0.0", forRemoval = true)
```

Comment:
Being invoked by some Spring Boot class even though our override doesn't invoke it. Not sure what's going on, but it's also causing a test failure for me.